### ISSUE TITLE
nomad fmt should return non zero exit code when changing files

### ISSUE THREAD

**Issue by jrnijboer** [hashicorp/nomad#16170]
### Nomad version
Nomad v1.4.3 (f464aca721d222ae9c1f3df643b3c3aaa20e2da7)

### Operating system and Environment details
VERSION="20.04.5 LTS (Focal Fossa)"
running under wsl2 in Windows 11

### Issue
There are 2 problems I have with nomad fmt. I'm not actually sure if they are bugs, or if they should be feature requests, so for now I'll call them "issues".

The first issue is that 'nomad fmt' returns a zero exit code when it has formatted files. I believe this should return a non zero exit code, this makes it easier to check if nomad files weren't properly formatted and have been changed (nice for pre-commit hooks). This is also how terraform fmt works. I think it would be nice for users to have some consistency between the Hashicorp products.

The other issue is that nomad fmt -check actually alters files (by formatting them). According to https://developer.hashicorp.com/nomad/docs/commands/fmt?product_intent=nomad#check this should only check the files if they are valid HCL. Running nomad fmt -check should not change files. However, I also believe this also should work more like terraform fmt -check, so not only should it check if the file is valid HCL, but also if it is properly formatted. So, return zero exit code if valid HCL and properly formatted, else return non zero exit code (when HCL invalid or not properly formatted) to make it consistent with other Hashicorp products (see [terraform fmt -check](https://developer.hashicorp.com/terraform/cli/commands/fmt#check)) 

### First issue (non zero exit code when nomad has changed files)
```
mkdir -p /tmp/nomad
rm -rf /tmp/nomad/*
git init
cat <<EOF >> job.hcl
job "nginx" {
    datacenters = ["dc1"]
    group "nginx" {
        task "container" {
            driver = "docker"
            config {
                image = "nginx-alpine"
            }
        }
    }
}
EOF
git add .
git status
nomad fmt job.hcl
echo "exit code: $?"
git status
```
the last git status shows that the file has been changed, but the exit code from nomad is zero

### Second issue (running with -check does change files)
```
mkdir -p /tmp/nomad
rm -rf /tmp/nomad/*
git init
cat <<EOF >> job.hcl
job "nginx" {
    datacenters = ["dc1"]
    group "nginx" {
        task "container" {
            driver = "docker"
            config {
                image = "nginx-alpine"
            }
        }
    }
}
EOF
git add .
git status
nomad fmt -check job.hcl
echo "exit code: $?"
git status
```
the last git status again shows that the file has been changed, this should not happen. But also the exit code from nomad is zero, this should be non zero because the file is not properly formatted.

**Comment by tgross** [comment]
Hi @jrnijboer! While you're right that the behavior is different than `terraform fmt`, in this case it's intentional. For Nomad we've found that for most of the commands where we're validating/checking something (ex. `nomad job plan`, `nomad job validate`), we've had equal numbers of users who really felt strongly that they want a non-zero exit code for diffs vs not. So for `nomad fmt` we decided to make that configurable by the user and so for the CI use case you've described, you'll want to use `nomad fmt -check`.

That being said, there's a bug fix that's landed in `main` for the `-check` exit code https://github.com/hashicorp/nomad/pull/15797. That's already available in the Nomad 1.5.0-beta.1 and will ship in a backport release to 1.4.x once 1.5.0 is GA (in the next few weeks).

There are two remaining bugs however, which are combining into the behavior you're seeing here:
* The `-write` flag is incorrectly documented to default to false.
* The `-check` flag doesn't exit early _before_ we write.

These are pretty trivial fixes... I'll try to get a PR up shortly.

**Comment by jrnijboer** [comment]
Thank you for your response @tgross. I think for the `-check` option the documentation should clarify that it not only checks for valid HCL, but if the HCL is valid it also checks for properly formatted HCL (which to me seems the whole point of a `fmt` command?). Unless valid HCL also implies that it's properly formatted?

Anyway, I just checked the 1.5 beta and that looks pretty suitable with the `-check` option for our CI runs. Looking forward for the 1.5 release!
